Source/cryptalgo/Hash.h

```diff
@@ -115,17 +131,33 @@ namespace Crypto {
         static const EnumHashType Type = HASH_SHA1;
         static const uint8_t Length = 20;
 
+        static uint32_t ContextLength()
+        {
+            return static_cast<uint32_t>(sizeof(_context));
+        }
+
+        inline const uint8_t* CurrentContext() const
+        {
+            return reinterpret_cast<const uint8_t*>(&_context);
+        }
+
+        inline void LoadContext(const uint8_t context[])
```

Comment:
As the constructore. make this type safe, you expect here a Context type, so set it on the interface like: inline void Load(const Context& contex); This way you also do not have to re-iteraet in the method name that it is a COntext as the parmater already depicts this. This makes it all type safe and more self-documenting.

Reply:
done